dfastmi/batch.py

```diff
@@ -139,56 +140,56 @@ def batch_mode_core(
             )
         )
     
-    branch = config["General"]["Branch"]
+    branch_name = config["General"]["Branch"]
     try:
-        ibranch = rivers.branches.index(branch)
+        ibranch = next((i for i, branch in enumerate(rivers.branches) if branch.name == branch_name), -1)
```

Comment:
would you say that this piece of code is very readable?

Reply:
Maybe a lil bit backwards, since you stated you didnt like all the dictionaries.
But might it be an Idea to make rivers.branches an dictionary of <branch_name, Branch> to improve accesability and performance?

just a thought and I am not sure yet if it is valid since this is the first part I see it being used with branch_name, so it might not be the best option.

Reply:
Why retrieve Ibranch as the index and not just return the branch?

Reply:
then later on you can do branch.reaches instead of rivers.branches[ibranch].reaches right?

Reply:
you already made a get reach method, can you reuse that here?

Reply:
might be tricky/risky since this is batch and those parts might not be covered, see the drawing in https://issuetracker.deltares.nl/browse/DFAST-103 regarding coverage.

Reply:
I agree, but batch.py is not in scope of this issue. I only refactor code to get it working again. Will pickup the good comments later in issue [DFAST-103](https://issuetracker.deltares.nl/browse/DFAST-103) or [DFAST-104](https://issuetracker.deltares.nl/browse/DFAST-104)

Reply:
Discussed, we will leave these comments for a later refactoring.
When a new issue is created for this, the number will be disclosed here.